apps/pipelines/nodes/base.py

```diff
@@ -23,71 +20,62 @@ def add_messages(left: list, right: list):
-
-    Define required parameters as typed fields. Compose the runnable in `get_runnable`
+    """Pipeline node that implements the `_process` method and returns a new state. Define required parameters as
+    typed fields.
 
     Example:
         class FunNode(PipelineNode):
             required_parameter_1: CustomType
             optional_parameter_1: Optional[CustomType] = None
 
-            def get_runnable(self, input) -> str:
-                if self.optional_parameter_1:
-                    return PromptTemplate.from_template(template=self.required_parameter_1)
-                else:
-                    return LLMResponse.build(node)
+            def _process(self, state: PipelineState) -> PipelineState:
+                input = state["messages"][-1]
+                output = ... # do something
+                return output # The state will be updated with output
 
        class FunLambdaNode(PipelineNode):
             required_parameter_1: str
 
-            def get_runnable(self, node: Node):
-                def fn(input: Input):
-                    return self.required_parameter_1
-                return RunnableLambda(fn, name=self.__class__.__name__)
+            def _process(self, state: PipelineState) -> PipelineState:
+                ...
+                return # The state will not be updated, since None is returned
+
     """
 
+    _config: RunnableConfig | None = None
+
     class Config:
         arbitrary_types_allowed = True
 
     @classmethod
     def build(cls, node: Node) -> Callable[[dict], dict]:
         try:
-            built_node = cls(**node.params)
+            return cls(**node.params)
         except ValidationError as ex:
             raise PipelineNodeBuildError(ex)
 
-        return built_node.get_runnable(node)
+    def process(self, state: PipelineState, config) -> PipelineState:
+        self._config = config
+        cls_name = self.__class__.__name__
+        self.logger.info(f"{cls_name} starting")
 
-    @classmethod
-    def get_callable(cls, node: Node) -> Callable:
-        built_node = cls.build(node)
+        output = self._process(state)
 
-        def fn(state: PipelineState) -> PipelineState:
-            if isinstance(built_node, BasePromptTemplate):
-                input = {"input": state["messages"][-1]}
-            else:
-                input = state["messages"][-1]
-            result = built_node.invoke(input)
-            if isinstance(result, BaseMessage):
-                output = result.content
-            elif isinstance(result, StringPromptValue):
-                output = result.text
-            else:
-                output = result
-            return PipelineState(messages=[output])
-
-        return fn
-
-    def get_runnable(self, node: Node) -> Runnable:
-        """Get a predefined runnable to be used in the pipeline"""
+        self.logger.info(f"{cls_name} finished with output: {output}")
```

Comment:
You probably saw in my other PR, but I started passing inputs and outputs as kwargs, otherwise the log page was getting too hard to read.